### PR TITLE
initial commit for ui-ux section

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
This PR adds the initial UI/UX section. Going to leave is WIP for now but I do encourage folks to start commenting on it if they wish. 

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [x] Label the pull request with the appropriate area(s)
- [x] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [x] Assign the project - Engineering Playbook Backlog
- [x] Assign the pull request to yourself

### CODE REVIEW

File: docs/ui-ux/README.md
Comment (KatieProchilo) [inline docs/ui-ux/README.md:37]: Looks like you're either missing a period or maybe meant to remove something from the last sentence

File: docs/ui-ux/README.md
Comment (KatieProchilo) [inline docs/ui-ux/README.md:35]: Remove last comma?

File: docs/ui-ux/README.md
Comment (KatieProchilo) [reply docs/ui-ux/README.md:37]: I couldn't find the "strict set of minimum requirements" by navigating to that link. Would these only come up once you download the browser tool below?

File: docs/ui-ux/README.md
Comment (KatieProchilo) [inline docs/ui-ux/README.md:49]: Missing links here, maybe part of the WIP 🙂

File: docs/ui-ux/README.md
Comment (KatieProchilo) [inline docs/ui-ux/README.md:52]: Not sure that my feedback will belong here since I love the casual/flexible/open-ended language you've used and this is just a README but...

I'd find it hugely helpful to point out the signs that maybe React is not the best choice and it's time for HTML instead or Web Components. Again, that seems a bit technical for the engineering playbook but would be interested to learn

File: docs/ui-ux/README.md
Comment (Ethan-Arrowood) [reply docs/ui-ux/README.md:37]: good catch. I rewrote this paragraph so it made more sense. You're right, there isn't exactly a "strict set of minimum requirements". But like every app needs to pass the accessibility insights tool. I'm going to go into much more detail on this in the accessibility page.

File: docs/ui-ux/README.md
Comment (Ethan-Arrowood) [reply docs/ui-ux/README.md:52]: Yeah i'm still pondering how to best author this topic. One comment I'm really happy with is in the beginning of this page: 

> Keep in mind that like all software, there is no "right way" to build a user interface application. Leverage and trust your team's or your customer's experience and expertise for the best development experience.

I feel like at the end of the day I want CSE crews to use what they are most comfortable with. What I don't want crews to do is to pick React because its what everyone else is using (insert a joke about friends and bridges here).

File: docs/ui-ux/README.md
Comment (engineersamuel) [inline docs/ui-ux/README.md:39]: Also I believe there is a node module or similar app which one can put as part of the CI/CD process to report out accessibility issues and how accessible a site is every time it is built.  Maybe look that up and include it here as well for automation.

File: docs/ui-ux/README.md
Comment (engineersamuel) [reply docs/ui-ux/README.md:49]: I may suggest rewording this point, for example:

- Typically, the default output for client-side JavaScript is sufficient for an MVP app, if running in production and targetting legacy browsers (which happens but is more uncommon) then look into polyfills and configuration such as (browserlist)[https://github.com/browserslist/browserslist]. 
- Typescript is recommended for new projects however JavaScript can be sufficient as well, but it lacks the robust Typing system of Typescript.

File: docs/ui-ux/README.md
Comment (Ethan-Arrowood) [reply docs/ui-ux/README.md]: That will be good for the accessibility document

**Comment by Ethan-Arrowood** [comment]
Publishing this PR as is so we can get it merged asap and then iterate on the incomplete sections. Hopefully after the M+C all hands I'll have more folks interested in authoring content too. @engineersamuel @KatieProchilo if y'all want to give this another look that would be appreciated. if you have additional content ideas maybe we can put them into an issue and include them in a follow up PR soon? I'd like to have a page to point to before Wednesday

File: docs/ui-ux/README.md
Comment (TessFerrandez) [inline docs/ui-ux/README.md:7]:
```suggestion
The goal of the **User Interface** section is to provide guidance on developing web applications for CSE engagements. Everyone should begin by reading the [General Guidance](#general-guidance) for a quick introduction to the four main aspects of every web application project. From there, readers are encouraged to dive deeper into each topic, or begin reviewing technical guidance that pertains to their engagement. All UI/UX projects should begin with a detailed design document. Review the [Design Document](#design-document) section for more details, and a template to get started.
```

We should see the playbook as standalone from the EF groups - and ideally shouldn't reference internal groups or processes. The playbook should ideally be readable and applicable also to teams outside of CSE.

File: docs/ui-ux/README.md
Comment (TessFerrandez) [inline docs/ui-ux/README.md:13]:
```suggestion
The state of web platform engineering is fast moving. There is no one-size-fits-all solution. For any team to be successful in building a UI, they need to have an understanding of the higher-level aspects of all UI project.
```

File: docs/ui-ux/README.md
Comment (TessFerrandez) [inline docs/ui-ux/README.md:26]: This refers to the internal game plan - can we rephrase this to make it more general?

File: docs/ui-ux/README.md
Comment (TessFerrandez) [inline docs/ui-ux/README.md:61]: Should this template be split out to a separate file, so teams can download and use it in their projects?

File: docs/ui-ux/recommended-technologies.md
Comment (TessFerrandez) [inline docs/ui-ux/recommended-technologies.md:20]:
```suggestion
React is a framework developed and maintained by Facebook. React is used throughout Microsoft and has a vast open source community.
```

**Review by TessFerrandez — APPROVED** [review]
Overall great start... I have approved the PR to not delay, but would like to see references to internal processes removed before the content is merged.

**Comment by Ethan-Arrowood** [comment]
Thank you for the great review @TessFerrandez - i've made all your recommended changes 😁

File: docs/ui-ux/README.md
Comment (wbreza) [inline docs/ui-ux/README.md:24]: It would be great to cross link into some of the "design" area document and work within them on any required revisions to the design docs area.

File: docs/ui-ux/README.md
Comment (wbreza) [inline docs/ui-ux/README.md:63]: Please have a discussion with @zmmille2 around merging int design related content if applicable.

**Review by wbreza — APPROVED** [review]
Great start.  Added some feedback around incorporating design requirements into the design EF area as neded.

File: docs/ui-ux/recommended-technologies.md
Comment (luisamiranda) [inline docs/ui-ux/recommended-technologies.md:3]: typo: "The purpose of this page is to review..."
typo: "To reiterate..."

File: docs/ui-ux/README.md
Comment (luisamiranda) [inline docs/ui-ux/README.md:46]: typo: "project to project" does not need hyphens.

**Review by luisamiranda — APPROVED** [review]
This is going to be really useful. Love the focus on accessibility and the template will be handy to standardize practices in CSE. I look forward to seeing future sections. I just pointed out a few typos.

File: docs/ui-ux/README.md
Comment (Ethan-Arrowood) [reply docs/ui-ux/README.md:24]: Great point. I'll make note of this and will make the modifications in a follow up. Thank you!

File: docs/ui-ux/README.md
Comment (Ethan-Arrowood) [reply docs/ui-ux/README.md:63]: Same as previous. I've also reached out to Zach to discuss.

**Comment by Ethan-Arrowood** [comment]
Thank you everyone who reviewed ❤️. I'm going to merge this once the CI passes. Other reviewers are welcome to continue commenting on this PR; I will be going back through tomorrow and opening issues for all of the work that needs to be done.

**Pull request merged**
